src/mining_minigame.c

```diff
@@ -0,0 +1,3395 @@
+                SetGpuReg(REG_OFFSET_BG3HOFS, 2);
+                SetGpuReg(REG_OFFSET_BG3VOFS, 2);
+                SetGpuReg(REG_OFFSET_BG2VOFS, 2);
+            }
+            sMiningUiState->shakeState++;
+            break;
+        case 12: // Right 1 - Down 1
+            MoveItemSprites(3, 3);
+            if (!IsStressLevelMax())
+                gSprites[sMiningUiState->ShakeHitEffect].invisible = 0;
+            gSprites[sMiningUiState->ShakeHitTool].x += 7;
+            StartSpriteAnim(&gSprites[sMiningUiState->ShakeHitTool], 1);
+            sMiningUiState->shakeState++;
+            break;
+        case 13:
+            if (sMiningUiState->toggleShakeDuringAnimation == FALSE)
+            {
+                SetGpuReg(REG_OFFSET_BG3HOFS, -1);
+                SetGpuReg(REG_OFFSET_BG2HOFS, -1);
+                SetGpuReg(REG_OFFSET_BG3VOFS, -1);
+                SetGpuReg(REG_OFFSET_BG2VOFS, -1);
+            }
+            sMiningUiState->shakeState++;
+            break;
+        case 15:
+            MoveItemSprites(-1, -1);
+            //if (!IsStressLevelMax())
+            //  gSprites[sMiningUiState->cursorSpriteIndex].invisible = 0;
+            sMiningUiState->shakeState++;
+            break;
+        case 16:
+            SetGpuReg(REG_OFFSET_BG3VOFS, 0);
+            SetGpuReg(REG_OFFSET_BG3HOFS, 0);
+            SetGpuReg(REG_OFFSET_BG2HOFS, 0);
+            SetGpuReg(REG_OFFSET_BG2VOFS, 0);
+            DestroySprite(&gSprites[sMiningUiState->ShakeHitTool]);
+            DestroySprite(&gSprites[sMiningUiState->ShakeHitEffect]);
+            if (sMiningUiState->shakeDuration > 0)
+            {
+                sMiningUiState->shakeDuration--;
+                sMiningUiState->shakeState = 0;
+                sMiningUiState->toggleShakeDuringAnimation = FALSE;
+                break;
+            }
+            if (IsStressLevelMax())
+                WallCollapseAnimation();
+            if (!IsStressLevelMax())
+                gSprites[sMiningUiState->cursorSpriteIndex].invisible = 0;
+            sMiningUiState->shakeState = 0;
+            sMiningUiState->shouldShake = FALSE;
+            sMiningUiState->toggleShakeDuringAnimation = FALSE;
+            DestroyTask(taskId);
+            break;
+        default:
+            sMiningUiState->shakeState++;
+            break;
+    }
+    BuildOamBuffer();
+}
+
+static void Mining_VBlankCB(void)
+{
+    Mining_CheckItemFound();
+    UpdatePaletteFade();
+    LoadOam();
+    ProcessSpriteCopyRequests();
+    TransferPlttBuffer();
+}
+
+static void Mining_FadeAndBail(void)
+{
+    BeginNormalPaletteFade(PALETTES_ALL, 0, 0, 16, RGB_BLACK);
+    CreateTask(Task_Mining_WaitFadeAndBail, 0);
+    SetVBlankCallback(Mining_VBlankCB);
+    SetMainCallback2(Mining_MainCB);
+}
+
+#define TILE_POS(x, y) (32 * (y) + (x))
+
+static void OverwriteTileDataInTilemapBuffer(u8 tile, u8 x, u8 y, u16* tilemapBuf, u8 pal)
+{
+    tilemapBuf[TILE_POS(x, y)] = tile | (pal << 12);
```

Comment:
```suggestion
static void OverwriteTileDataInTilemapBuffer(u8 tile, u8 x, u8 y, u16* tilemapBuf, u8 pal)
{
    tilemapBuf[32 * y + x] = tile | (pal << 12);
```
The macro was only used once, so it's not really needed.